frame/support/procedural/src/construct_runtime/mod.rs

```diff
@@ -163,6 +229,12 @@ fn decl_outer_inherent<'a>(
 		})
 	});
 	quote!(
+		// Prevent UncheckedExtrinsic to print unused warning.
+		const _: () = {
```

Comment:
I see this pattern in the expansions of derive macros of codec as well, what is the purpose? is the keyword `__hidden_use_unchecked_extrinsic ` used anywhere now?

Reply:
not but this is a pattern used by may derive macro, for instance serde.
I think the main advantage is that it makes `type __hidden_use_unchecked_extrinsic` unavailable in doc and rust error will not recommand it.

Reply:
But if the type alias is not even accessible and used anywhere, I don't see how it is used. 

For example I've this and it makes sense: 

```rust
const _: () { impl Something for SomethingElse { ... } } 
``` 

but something like: 

```rust
const _: () = type __unused_alias = Foo; 
```

I don't get. 

Unless is `__hidden_use_unchecked_extrinsic` used in the other branch? Asking because I couldn't find any usage here.

Reply:
this is just to `// Prevent UncheckedExtrinsic to print unused warning.`
so I have `const _: () = { #[allow(unused)] type __unused_alias = Foo; }`
UncheckedExtrinsic is unused if there is no inherent.